Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -632,25 +632,38 @@ function Add-AzureSqlDatabaseServerFirewallRule
           [String] [Parameter(Mandatory = $true)] $firewallRuleName)
     
     Trace-VstsEnteringInvocation $MyInvocation
-    Write-Verbose "Creating firewall rule $firewallRuleName"
 
-    $connectionType = Get-ConnectionType -serviceEndpoint $endpoint
-
-    if(IsLegacyAzureConnection $connectionType)
-    {
-        Add-LegacyAzureSqlServerFirewall -endpoint $endpoint -serverName $serverName -startIPAddress $startIPAddress -endIPAddress $endIPAddress -firewallRuleName $firewallRuleName
-    }
-    elseif (IsAzureRmConnection $connectionType)
+    try
     {
+        Write-Verbose "Creating firewall rule $firewallRuleName"
+
+        $connectionType = Get-ConnectionType -serviceEndpoint $endpoint
 
-        Add-AzureRmSqlServerFirewall -endpoint $endpoint -serverName $serverName -startIPAddress $startIPAddress -endIPAddress $endIPAddress -firewallRuleName $firewallRuleName
+        if(IsLegacyAzureConnection $connectionType)
+        {
+            Add-LegacyAzureSqlServerFirewall -endpoint $endpoint -serverName $serverName -startIPAddress $startIPAddress -endIPAddress $endIPAddress -firewallRuleName $firewallRuleName
+        }
+        elseif (IsAzureRmConnection $connectionType)
+        {
+
+            Add-AzureRmSqlServerFirewall -endpoint $endpoint -serverName $serverName -startIPAddress $startIPAddress -endIPAddress $endIPAddress -firewallRuleName $firewallRuleName
+        }
+        else
+        {
+            throw (Get-VstsLocString -Key AZ_UnsupportedAuthScheme0 -ArgumentList $connectionType)
+        }
+
+        Write-Verbose "Firewall rule $firewallRuleName created"
     }
-    else
+    catch 
     {
-        throw (Get-VstsLocString -Key AZ_UnsupportedAuthScheme0 -ArgumentList $connectionType)
+        $parsedException = Parse-Exception($_.Exception)
```

Comment:
$parsedException will be null only when $._Exception will be null , in that Case (throw $_.Exception.ToString()) in next line will in itself be an NPE , so its not required , right ??

Also is it possible that $_.exception will be null , if not than we should remove null check in parsed Exception function and also remove if() in next line.